### PR TITLE
Fixes revenants getting irradiated

### LINKED ISSUE #52078: revenants can invisible own people with radiation
## Round ID:
[141373](https://scrubby.melonmesa.com/round/141373)

## Testmerges:
#51912: 'Painting Cleanup V2' by AnturK at commit 3e9aad85cd

## Reproduction:

-be revenant
-be near supermatter
-become irradiated and able to irradiate others

might apply to slaughter demons as well. very much so not a fan of an antagonist where there is literally nothing you can do about them  until they attack but if they can irradiate themselves and passively hurt you then they can attack while being invincible

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
fix: Revenants are no longer affected by radiation.
fix: Revenants can no longer murder the hell out of people with their radiation.
/:cl:

Fixes #52078

### CODE REVIEW

**Comment by ShizCalev** [comment]
one does not simply irradiate a ghost.

**Comment by carshalash** [comment]
Danny phantom glow.

**Comment by LemonInTheDark** [comment]
Damnit shiz you're too fast

**Comment by LemonInTheDark** [comment]
Does this actually prevent contamination? TRAIT_RADIMMUNE doesn't appear to effect it

File: code/modules/antagonists/revenant/revenant.dm [hunk 1]
```diff
@@ -85,6 +86,12 @@
 /mob/living/simple_animal/revenant/canUseTopic(atom/movable/M, be_close=FALSE, no_dexterity=FALSE, no_tk=FALSE)
 	return FALSE
 
+/mob/living/simple_animal/revenant/apply_effect(effect, effecttype, blocked)
+	if(effecttype == EFFECT_IRRADIATE)
+		return FALSE
+	return ..()
```
Comment (ShizCalev) [inline code/modules/antagonists/revenant/revenant.dm:92]: this isn't needed if #52090 gets merged

**Comment by ShizCalev** [comment]
@LemonInTheDark see #52090

**Comment by LemonInTheDark** [comment]
Does it handle contamination?

**Pull request closed without merging**